app/src/main/java/woowacourse/shopping/ShoppingProvider.kt

```
@@ -0,0 +1,22 @@
+package woowacourse.shopping
+
+import woowacourse.shopping.data.product.ProductRepository
+import woowacourse.shopping.data.shoppingcart.ShoppingCartRepository
+
+object ShoppingProvider {
+    private var _shoppingCartRepository: ShoppingCartRepository? = null
+
+    val shoppingCartRepository: ShoppingCartRepository get() = _shoppingCartRepository ?: throw IllegalArgumentException()
```

detekt (1.23.8)

[warning] A call to the default constructor of an exception was detected. Instead one of the constructor overloads should be called. This allows to provide more meaningful exceptions.

(detekt.exceptions.ThrowingExceptionsWithoutMessageOrCause)